fairseq/models/speech_to_text/xm_transformer.py

```diff
@@ -623,15 +640,10 @@ def build_model(cls, args, task):
 
         # make sure all arguments are present in older models
         base_architecture(args)
-        if getattr(args, "load_pretrained_decoder_from", None):
-            ckpt = torch.load(getattr(args, "load_pretrained_decoder_from", None))
-            decoder_args_dict = cls.get_decoder_args_from_checkpoint(ckpt["cfg"])
-            args = cls.override_decoder_args(args, decoder_args_dict)
-
-        def build_embedding(dictionary, embed_dim):
-            num_embeddings = len(dictionary)
-            padding_idx = dictionary.pad()
-            return Embedding(num_embeddings, embed_dim, padding_idx)
+        # if getattr(args, "load_pretrained_decoder_from", None) is not None:
+        #     ckpt = torch.load(getattr(args, "load_pretrained_decoder_from", None))
+        #     decoder_args_dict = cls.get_decoder_args_from_checkpoint(ckpt["cfg"])
+        #     args = cls.override_decoder_args(args, decoder_args_dict)
```

Comment:
This one is commented now, because currently we want to override some of the pretrained mbart parameters: we remove learned pos emb and layernorm from the embedding of that model. However, turning this off completely may lead to unintentional problems later when we forgot to match all the mbart parameters from pretrained model which are different from the default. 
I wonder if we could utilize model-override arg here somehow to make sure that we can do both loading the cfg from the pretrained mbart, and then add / change it with args from the CLI.

Reply:
@uralik @sravyapopuri388 Thank you for your feedback. I reverted the change because it doesn't affect two-pass models now.